src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -56,6 +56,29 @@ public static bool TryDecode(ReadOnlySpan<byte> utf8, Span<char> utf16, out int
             bytesConsumed = 0;
             charactersWritten = 0;
 
+            if (Vector.IsHardwareAccelerated)
+            {
+                ref var utf8Vectors = ref utf8.NonPortableCast<byte, Vector<byte>>().DangerousGetPinnableReference();
+                ref var utf16Vectors = ref utf16.NonPortableCast<char, Vector<ushort>>().DangerousGetPinnableReference();
+
+                int asciiLoops = Math.Min(utf8.Length, utf16.Length) / Vector<byte>.Count, asciiLoop = 0, outputOffset = 0;
```

Comment:
In *this* case, I need to be careful to account for the fact that utf16 might not be long enough - will see if I can do something suitable with slice instead - which might leave the loop open to elide (if the JIT does that)